crates/lib3h/src/engine/ghost_engine.rs

```diff
@@ -782,11 +782,11 @@ pub fn handle_gossip_to<
         //            }
         //            // TODO END
 
-        // Convert DHT Gossip to P2P Gossip
+        // Convert DHT *GossipTo* to P2P *Gossip*
         let p2p_gossip = P2pProtocol::Gossip(GossipData {
             space_address: gateway_identifier.clone(),
             to_peer_name: to_peer_name.clone(),
-            from_peer_name: Lib3hUri::with_undefined(), // FIXME
+            from_peer_name: Lib3hUri::with_undefined(), // Should be gateway.as_mut().as_mut().this_peer().peer_location,
```

Comment:
Is this breaking any functionality or is from_peer_name not a super important field?

Reply:
Not yet it seems but it should be a breaking functionality otherwise the field is useless and should be removed.